### DISCUSSION TITLE
"Let's not"

### DISCUSSION THREAD

**Discussion by glaucocustodio** [evenbetterspecs/evenbetterspecs.github.io#14] [General]
What are your thoughts on https://evenbetterspecs.github.io/#lets-not?

**Comment by pdmholden** [comment]
I understand that the 'distance' between the declaration of `user` and the expectation makes the spec harder to read. What I don't understand is what role `#let` plays in this and why `#let` itself is not recommended.

As an alternative, does this not accomplish the same goal?

```ruby
describe '#full_name' do
  context 'when first name and last name are present' do
    let(:user) { build(:user, first_name: 'Edson', last_name: 'Pelé') }

    it 'returns the full name' do
      expect(user.full_name).to eq('Edson Pelé')
    end
  end

  context 'when last name is not present' do
    let(:user) { build(:user, first_name: 'Edson', last_name: nil) }

    it 'returns the first name' do
      expect(user.full_name).to eq('Edson')
    end
  end
end
```

**Reply by glaucocustodio** [reply]
- makes the tests less self-contained 
- requires people to know what `let`/`let!` does (other test frameworks don't have such concept) 
- requires more typing
 
I prefer to keep things stupidly obvious and simple.
